[WIP] Flightmode improvements

philippotto opened the pull request:
### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- abc

### Issues:
- fixes #123

------
- [ ] Updated [changelog](../blob/master/CHANGELOG.md#unreleased)
- [ ] Updated [migration guide](../blob/master/MIGRATIONS.md#unreleased) if applicable
- [ ] Updated [documentation](../blob/master/docs) if applicable
- [ ] Needs datastore update after deployment
- [ ] Ready for review

daniel-wer commented:
I did a quick test after seeing the discuss post and encountered a large grey circle that doesn't go away, even after a reload. Maybe you can repro it: https://flightmodeimprovement.webknossos.xyz/annotations/Explorational/5bd040530100009e02ca258b#42046,18988,1725,1,1.24,99.95,84.82,267.14,152

philippotto commented:
> I did a quick test after seeing the discuss post and encountered a large grey circle that doesn't go away, even after a reload. Maybe you can repro it: https://flightmodeimprovement.webknossos.xyz/annotations/Explorational/5bd040530100009e02ca258b#42046,18988,1725,1,1.24,99.95,84.82,267.14,152

Thanks for testing! The grey circle is outside the bounding box of the existing data. So, it's correct that nothing is rendered. However, the circle should actually be black, since gray indicates "is loading". In my PR, I changed the look up logic a bit, which is why the "display black if buckets are missing inside the bounding box"-logic does not work anymore. I have to figure out a way to fix this again. I'll add a todo item to this PR!

philippotto commented:
@daniel-wer I cleaned up the code quite a bit, but it's not really polished yet. Nevertheless, I'd like to get some feedback from you on this PR first. I updated the PR description with more details.

In addition to overall polishing, I'd like to share more code between the TD view implementation of ortho and arbitrary mode. Also, it still need to re-think how we want to render missing data black while still being able to render fallback data, in case the necessary buckets just weren't selected properly. One option which might work is to also pass null buckets to the TextureBucketManager which then just marks a `-4` or similar in the look up texture without actually uploading data.

philippotto commented:
I managed to extract the duplicated code for the td controller. I think I'll do the rest tomorrow to get a clear head :)

philippotto commented:
@daniel-wer For merging I need an accept for this PR :) Fyi, I created #3446 as a follow-up. Gray vs. black data is rendered in this PR correctly for the ortho mode (which is probably where it's most important), but not yet for flight mode. I think, this should be acceptable though for now.
